src/main/scala/org/constellation/infrastructure/redownload/RedownloadPeriodicCheck.scala

```diff
@@ -0,0 +1,27 @@
+package org.constellation.infrastructure.redownload
+
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import org.constellation.util.Logging.logThread
+import org.constellation.{ConstellationExecutionContext, DAO}
+import org.constellation.util.PeriodicIO
+
+import scala.concurrent.duration._
+
+class RedownloadPeriodicCheck(periodSeconds: Int = 30)(implicit dao: DAO)
+    extends PeriodicIO("RedownloadPeriodicCheck") {
+
+  val contextShift: ContextShift[IO] = IO.contextShift(ConstellationExecutionContext.bounded)
+
+  private def triggerRedownloadCheck(): IO[Unit] =
+    for {
+      _ <- dao.redownloadService.fetchPeersProposals()
```

Comment:
Just a note, let's make sure that we're chunk serializing these Snapshots, otherwise we could see kryo buffer errors like before